packages/core/src/types.ts

```diff
@@ -0,0 +1,23 @@
+import type { TypedDataDomain, TypedDataField } from 'ethers'
+import type { PatchedRequestInit } from 'graphql-request/dist/types'
+
+export type ClientOption = {
+  endpoint: URL
+  options?: PatchedRequestInit
+}
+
+export type EIP712Data = {
+  domain: TypedDataDomain
+  types: Record<string, Array<TypedDataField>>
+  message: Record<string, any>
+}
+
+export type TransactionHash = `0x${string}`
+export type ChainId = number
+export type Address = `0x${string}`
+export type UUID = string
+
+export interface IAction<T = string, V = any> {
```

Comment:
Those are not actions, they represent a new internal state of an action.
Also I think generic can be more than one letter and will be good to provide more context:
```suggestion
export interface IState<Type = string, Payload = any> {
```

Reply:
[825d8e8](https://github.com/liteflow-labs/liteflow-js/pull/142/commits/825d8e8274c1a7a33ce44bceaa79336033c69846)